storage/innobase/handler/ha_innodb.cc

```diff
@@ -16452,14 +16452,15 @@ ha_innobase::external_lock(
 
 		if (UNIV_UNLIKELY(trx->take_stats)) {
 			increment_thd_innodb_stats(thd,
-						   (unsigned long long) trx->id,
+						   (unsigned long long) trx->id? trx->id: trx->id_saved,
```

Comment:
Please put the ternary expression into parens (or the type cast applies to trx->id only)
No spaces before ? and :
Since you are touching this line, please use C++ style cast